File: leetcode3/최원준/3014. Minimum Number of Pushes to Type Word I.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        counter = Counter(word)
        freq_num = [[freq, num] for num, freq in counter.items()]
        freq_num.sort()
        
        ans = 0
        counter = 0

        while freq_num:
            freq, num = freq_num.pop()
            cost = counter // 8 + 1
            counter+=1
            ans+=cost * freq
        return ans

File: leetcode3/황은지/3014. Minimum Number of Pushes to Type Word I.js
/**
 * @param {string} word
 * @return {number}
 */
var minimumPushes = function(word) {
    // 최대한 한번만 누르게 -> 그래도 부족한건 2개, 3개짜리로..
    let cost=0;
    let click=1;
    let left=word.length;

    while(left>0){
        if(left<=8){
            cost+=(left*(click++))
            break;
        }else{
            left-=8;
            cost+=(8*(click++));
        }
    }

    return cost;
};